src/components/InteractionLayer/index.js

```diff
@@ -462,24 +448,76 @@ class InteractionLayer extends React.Component {
         newPoints.push({ id: s.id });
       }
     });
+    return newPoints;
+  };
 
-    if (ruler && ruler.visible) {
-      this.setState({ points: newPoints });
+  setRulerPosition = timestamp => {
+    if (!timestamp) {
+      this.setState({
+        points: [],
+        touchX: null,
+        touchY: null,
+      });
+      return;
+    }
+    const { xScalerFactory, width, xSubDomain } = this.props;
+    const xScale = xScalerFactory(xSubDomain, width);
+    const xpos = xScale(timestamp);
+    this.setRulerPoints(xpos);
+    this.setState({
+      touchX: xpos,
+    });
+  };
+
+  setRulerPoints = xpos => {
+    const { ruler, width, xSubDomain, xScalerFactory } = this.props;
+    if (!ruler || !ruler.visible) {
+      return [];
     }
+    const xScale = xScalerFactory(xSubDomain, width);
+    const rawTimestamp = xScale.invert(xpos).getTime();
+    const rulerPoints = this.getRulerPoints(rawTimestamp);
+    this.setState({ points: rulerPoints });
 
+    return rulerPoints;
+  };
+
+  setArea = (xpos, ypos) => {
     const { area } = this.state;
-    if (area) {
-      const output = this.getDataForCoordinate(xpos, ypos, true);
-      this.setState({
-        area: {
-          ...area,
-          end: output,
-        },
-      });
+    if (!area) {
+      return;
     }
+    const output = this.getDataForCoordinate(xpos, ypos, true);
+    this.setState({
+      area: {
+        ...area,
+        end: output,
+      },
+    });
+  };
+
+  processExternalMouseMove = (xpos, ypos) => {
+    this.processMouseMove(xpos, ypos, true);
+  };
 
+  processMouseMove = (xpos, ypos, external = false) => {
```

Comment:
oh boy, `external: boolean` seems _dangerous_. Why not pass the actual event in?